OSDOCS-16607 [ROSA]: Vale and Dita work for Networking (#110996)
openshift/openshift-docs: modules/configuring-a-proxy-after-installation-cli.adoc

[id="configuring-a-proxy-after-installation-cli_{context}"]
= Configuring a proxy after installation using the CLI

[role="_abstract"]
You can use the ROSA CLI (`rosa`) to add a cluster-wide proxy configuration to an existing ROSA cluster in a Virtual Private Cloud (VPC).

You can also use `rosa` to update an existing cluster-wide proxy configuration. For example, you might need to update the network address for the proxy or replace the additional trust bundle if any of the certificate authorities for the proxy expire.

[IMPORTANT]
====
The cluster applies the proxy configuration to the control plane and compute nodes. While applying the configuration, each cluster node is temporarily placed in an unschedulable state and drained of its workloads. The process restarts each node.
====

.Prerequisites

* You have installed and configured the latest ROSA (`rosa`) and OpenShift (`oc`) command-line interfaces (CLIs) on your installation host.

> * You deploy your {product-title} cluster in a VPC.

Reviewer: ```suggestion
* You have deployed your {product-title} cluster in a VPC.
```
Could help keep list items parallel.

.Procedure

> . Edit the cluster configuration to add or update the cluster-wide proxy details:

Reviewer: ```suggestion
* Edit the cluster configuration to add or update the cluster-wide proxy details:
```
One-step procedures should be unnumbered lists.

.Procedure

* Edit the cluster configuration to add or update the cluster-wide proxy details:
. Edit the cluster configuration to add or update the cluster-wide proxy details:
+
[source,terminal]
----
$ rosa edit cluster \
 --cluster $CLUSTER_NAME \
 --additional-trust-bundle-file <path_to_ca_bundle_file> \ <1> <2> <3>
 --http-proxy http://<username>:<password>@<ip>:<port> \ <1> <3>
 --https-proxy https://<username>:<password>@<ip>:<port> \ <1> <3>
  --no-proxy example.com <4>
 --additional-trust-bundle-file <path_to_ca_bundle_file> \
 --http-proxy http://<username>:<password>@<ip>:<port> \
 --https-proxy https://<username>:<password>@<ip>:<port> \
  --no-proxy example.com
----
+

> --

Reviewer: There's now the OCP docs [callout replacement guide](https://docs.google.com/document/d/1j78SA8Y-ZRlVATbVe6D8bdYj7mQzzhOJtGt1xH_3qa0) with examples, and there's [Andrea's PR with examples](https://github.com/openshift/openshift-docs/pull/102276/changes).

In short, we look at 3 options:

- Use a simple sentence (for single callouts/commands)
- Use a description list
- Use a bulleted list

Would you please reformat the lines below into one of the lists?

----
$ rosa edit cluster \
 --cluster $CLUSTER_NAME \
 --additional-trust-bundle-file <path_to_ca_bundle_file> \
 --http-proxy http://<username>:<password>@<ip>:<port> \
 --https-proxy https://<username>:<password>@<ip>:<port> \
  --no-proxy example.com
----
+
--
** The `additional-trust-bundle-file`, `http-proxy`, and `https-proxy` arguments are all optional.

> **  The `additional-trust-bundle-file` argument is a file path pointing to a bundle of PEM-encoded X.509 certificates, which are all concatenated together. The additional-trust-bundle-file argument is a file path pointing to a bundle of PEM-encoded X.509 certificates, which are all concatenated together. The additional-trust-bundle-file argument is required for users who use a TLS-inspecting proxy unless the identity certificate for the proxy is signed by an authority from the {op-system-first} trust bundle. This applies regardless of whether the proxy is transparent or requires explicit configuration using the `http-proxy` and `https-proxy` arguments.

Reviewer 2: i thought we were using the definition list for callout replacements e.g.

```suggestion
where:

<path_to_ca_bundle_file> :: The file path pointing to a bundle of PEM-encoded X.509 certificates, which are all concatenated together. The additional-trust-bundle-file argument is a file path pointing to a bundle of PEM-encoded X.509 certificates, which are all concatenated together. The additional-trust-bundle-file argument is required for users who use a TLS-inspecting proxy unless the identity certificate for the proxy is signed by an authority from the {op-system-first} trust bundle. Otherwise, this field is optional. This applies regardless of whether the proxy is transparent or requires explicit configuration using the `http-proxy` and `https-proxy` arguments.
```
is this something different?

+
[IMPORTANT]
====
Do not attempt to change the proxy or additional trust bundle configuration on the cluster directly. Any changes must be applied by using the ROSA CLI (`rosa`) or {cluster-manager-first}. Any changes made directly to managed resources on the cluster are reverted automatically.
====
** The `http-proxy` and `https-proxy` arguments must point to a valid URL.

> ** A comma-separated list of destination domain names, IP addresses, or network CIDRs to exclude proxying.

Reviewer 2: see above callout list rewrite comment

+
[IMPORTANT]
====
Do not try to change the proxy or additional trust bundle configuration on the cluster directly. You must apply any changes by using the ROSA CLI (`rosa`) or {cluster-manager-first}. The system automatically reverts any changes made directly to managed resources on the cluster.
====
+
The `http-proxy` and `https-proxy` arguments must point to a valid URL.
+
The `no-proxy` argument accepts a comma-separated list of destination domain names, IP addresses, or network Classless Inter-Domain Routings (CIDRs) to exclude proxying.
+
Preface a domain with `.` to match subdomains only. For example, `.y.com` matches `x.y.com`, but not `y.com`. Use `*` to bypass proxy for all destinations.
+
If you scale up workers that are not included in the network defined by the `networking.machineNetwork[].cidr` field from the installation configuration, you must add them to this list to prevent connection issues.
+

> The system ignores this field if you do not set the `httpProxy` or `httpsProxy` fields.

Reviewer: In previews this is rendered as 2 description list items which can be confusing to users. It probably will make sense to have 4 items as it currently is in the published content (please see screenshots).

<img width="550" height="776" alt="Image" src="https://github.com/user-attachments/assets/1d568d29-5476-499a-95ea-43e0ca1c797e" /> 

<img width="586" height="627" alt="Image" src="https://github.com/user-attachments/assets/2aeadc78-0dcb-4973-99f6-8a5055d43f20" />

Author: @otikhomi thank you! Just to be clear, I _added back_ the callouts as that seemed to be what you wanted. Just let me know if I misunderstood :)

Reviewer: @jc-berger Sorry if my comment was confusing. We can't have callouts :) We'll have to opt for either a description list or unnumbered bullet points.

By adding the screenshots I hoped to illustrate how the published doc has 4 paragraphs (list items), and the PR has 2. So in the PR it looks like most of the content is about the `additional-trust-bundle-file` argument while half of it isn't related.

So in a PR we'd need to either have 4 description list items or 4 bulleted list items. Does that make sense?

Reviewer: Arrows mark where we need to have separate list items.

So instead of attaching all content with + we want to have a separate `title:: description` pair for a description list. 
Or we want to have 4 bulleted list items.

<img width="598" height="632" alt="Screenshot 2026-05-07 at 1 29 18 PM" src="https://github.com/user-attachments/assets/8eeb0233-834a-4818-b551-9ea1466b5f26" />

Author: +
--
** The `additional-trust-bundle-file`, `http-proxy`, and `https-proxy` arguments are all optional.
** The `additional-trust-bundle-file` argument is a file path pointing to a bundle of PEM-encoded X.509 certificates, which are all concatenated together. The additional-trust-bundle-file argument is required for users who use a TLS-inspecting proxy unless the identity certificate for the proxy is signed by an authority from the {op-system-first} trust bundle. This applies regardless of whether the proxy is transparent or requires explicit configuration using the http-proxy and https-proxy arguments.
** The `http-proxy` and `https-proxy` arguments must point to a valid URL.
**  A comma-separated list of destination domain names, IP addresses, or network CIDRs to exclude proxying.
+
Preface a domain with `.` to match subdomains only. For example, `.y.com` matches `x.y.com`, but not `y.com`. Use `*` to bypass proxy for all destinations.
If you scale up workers that are not included in the network defined by the `networking.machineNetwork[].cidr` field from the installation configuration, you must add them to this list to prevent connection issues.
+
This field is ignored if neither the `httpProxy` or `httpsProxy` fields are set.
--
